addons/minio/2020-01-25T02-50-51Z/install.sh

```diff
@@ -23,6 +33,18 @@ function minio() {
 }
 
 function minio_creds() {
+    # in the case of using a "hostPath", minio will generate a ".minio.sys" directory
+    # that is used to control access to that hostPath. this can be a problem when that hostpath
+    # is a directory in a shared file system (an NFS mount for example) and the installer is being
+    # run on a fresh instance, because new credentials will be generated and minio won't be able access the old data,
+    # so we make sure that ".minio.sys" is regenerated from the current minio credentials.
+    if [ -n "$MINIO_HOSTPATH" ]; then
+        # reset minio server access creds
+        rm -rf "$MINIO_HOSTPATH/.minio.sys"
```

Comment:
@salahalsaleh Is this going to remove that directory every time kurl install is re-run? That seems a bit heavy-handed and dangerous.

Are we positive there are no negative consequences for wiping out this directory? What if someone had set up Minio manually on that same path, and used encryption? I know we don't support encryption in the kURL addon, but the NFS mount point could easily be outside kURL control.

Is it possible to detect if the .minio.sys directory is compatible with the current instance, and only delete if necessary? Perhaps with user confirmation as well?

Reply:
technically, only removing the `config` directory inside `.minio.sys` is required, and both are regenerated if minio starts up and couldn't find them. i can update the code to only delete the `config` directory.

in case another minio instance was set up manually with server side encryption, deleting this directory will prevent that minio instance from accessing the data (we don't delete the actual data).

user confirmation logic sounds like a good idea.